odoo/orm/domains.py

```diff
@@ -186,7 +186,7 @@ def __new__(cls, *args):
         a field (str), the operator (str) and a value for the condition.
         """
         if len(args) > 1:
-            if isinstance(args[0], str):
+            if isinstance(args[0], str | IdentifierBuilder):
```

Comment:
`str | IdentifierBuilder` is a wrong abstraction. Even `str` is a poor abstraction. We should have a `FieldExpression` class that can be a field, accessor of a field, custom, etc.
I have a small draft on branch master-field-expression-krma.

If you have an IdentifierBuilder as a field_expr, you should use the CustomDomain directly.